server/src/main/java/io/crate/protocols/postgres/AuthenticationContext.java

```diff
@@ -83,8 +85,8 @@ SecureString password() {
      */
     @Override
     public void close() {
-        if (password != null) {
-            password.close();
+        if (credentials.password() != null) {
+            credentials.password().close();
         }
```

Comment:
Can we implement `Closable` on the `Credentials` class and close it instead? (e.g. if the SecureString token will be stored into a dedicated var at the `Credentials` class it should be closed as well.
```suggestion
        if (credentials != null) {
            credentials.close();
        }
```